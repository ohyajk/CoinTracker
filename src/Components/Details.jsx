import React from 'react';
import { useSelector } from 'react-redux';

const Details = () => {
  const details = useSelector((state) => state.coins.selectedCoin);

  return (
    <div data-testid="Details-1" className="flex flex-col gap-2 justify-start items-between mt-10 [&>*:nth-child(2n+3)]:bg-violet-900/20 border border-black/10">
      <img className="h-20 w-max ml-auto mr-auto" src={details.icon} alt={details.name} />
      <div className="flex justify-between px-4 py-4">
        <span>Name :</span>
        <span>
          {details.name}
          <i className="fa-solid fa-circle-arrow-right pl-2" />
        </span>
      </div>
      <div className="flex justify-between px-4 py-4">
        <span> Rank :</span>
        <span>
          {details.rank}
          <i className="fa-solid fa-circle-arrow-right pl-2" />
        </span>
      </div>
      <div className="flex justify-between px-4 py-4">
        <span>Total Supply :</span>
        <span>
          {details.totalSupply}
          <i className="fa-solid fa-circle-arrow-right pl-2" />
        </span>
      </div>
      <div className="flex justify-between px-4 py-4">
        <span>Available Supply :</span>
        <span>
          {details.availableSupply}
          <i className="fa-solid fa-circle-arrow-right pl-2" />
        </span>
      </div>
      <div className="flex justify-between px-4 py-4">
        <span>Price :</span>
        <span>
          {details.price}
          <i className="fa-solid fa-circle-arrow-right pl-2" />
        </span>
      </div>
      <div className="flex justify-between px-4 py-4">
        <span> Price Change Weekly :</span>
        <span>
          {details.priceChange1w < 0 ? <span className="text-red-400">{details.priceChange1w}</span> : <span className="text-green-400">{details.priceChange1w}</span>}
          <i className="fa-solid fa-circle-arrow-right pl-2" />
        </span>
      </div>
      <div className="flex justify-between px-4 py-4">
        <span>Price Change Daily :</span>
        <span>
          {details.priceChange1d < 0 ? <span className="text-red-400">{details.priceChange1d}</span> : <span className="text-green-400">{details.priceChange1d}</span>}
          <i className="fa-solid fa-circle-arrow-right pl-2" />
        </span>
      </div>
      <div className="flex justify-between px-4 py-4">
        <span>Price Change Hourly :</span>
        <span>
          {details.priceChange1h < 0 ? <span className="text-red-400">{details.priceChange1h}</span> : <span className="text-green-400">{details.priceChange1h}</span>}
          <i className="fa-solid fa-circle-arrow-right pl-2" />
        </span>
      </div>
    </div>
  );
};

export default Details;
