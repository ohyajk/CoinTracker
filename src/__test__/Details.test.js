import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../Redux/ConfigureStore';
import Details from '../Components/Details';
import '@testing-library/jest-dom/extend-expect';

test('should render Details', () => {
  const tree = render(
    <Provider store={store}>
      <BrowserRouter>
        <Details />
      </BrowserRouter>
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});
test('should render Rank', () => {
  render(
    <Provider store={store}>
      <BrowserRouter>
        <Details />
      </BrowserRouter>
    </Provider>,
  );
  expect(screen.getByTestId('Details-1')).toHaveTextContent('Rank');
});
