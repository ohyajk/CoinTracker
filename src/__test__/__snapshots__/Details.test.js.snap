// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Details 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-col gap-2 justify-start items-between mt-10 [&>*:nth-child(2n+3)]:bg-[#d1447a]"
        data-testid="Details-1"
      >
        <img
          class="h-20 w-max ml-auto mr-auto"
        />
        <div
          class="flex justify-between px-4 py-4"
        >
          <span>
            Name :
          </span>
          <span>
            <i
              class="fa-solid fa-circle-arrow-right pl-2"
            />
          </span>
        </div>
        <div
          class="flex justify-between px-4 py-4"
        >
          <span>
             Rank :
          </span>
          <span>
            <i
              class="fa-solid fa-circle-arrow-right pl-2"
            />
          </span>
        </div>
        <div
          class="flex justify-between px-4 py-4"
        >
          <span>
            Total Supply :
          </span>
          <span>
            <i
              class="fa-solid fa-circle-arrow-right pl-2"
            />
          </span>
        </div>
        <div
          class="flex justify-between px-4 py-4"
        >
          <span>
            Available Supply :
          </span>
          <span>
            <i
              class="fa-solid fa-circle-arrow-right pl-2"
            />
          </span>
        </div>
        <div
          class="flex justify-between px-4 py-4"
        >
          <span>
            Price :
          </span>
          <span>
            <i
              class="fa-solid fa-circle-arrow-right pl-2"
            />
          </span>
        </div>
        <div
          class="flex justify-between px-4 py-4"
        >
          <span>
             Price Change Weekly :
          </span>
          <span>
            <span
              class="text-green-400"
            />
            <i
              class="fa-solid fa-circle-arrow-right pl-2"
            />
          </span>
        </div>
        <div
          class="flex justify-between px-4 py-4"
        >
          <span>
            Price Change Daily :
          </span>
          <span>
            <span
              class="text-green-400"
            />
            <i
              class="fa-solid fa-circle-arrow-right pl-2"
            />
          </span>
        </div>
        <div
          class="flex justify-between px-4 py-4"
        >
          <span>
            Price Change Hourly :
          </span>
          <span>
            <span
              class="text-green-400"
            />
            <i
              class="fa-solid fa-circle-arrow-right pl-2"
            />
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-col gap-2 justify-start items-between mt-10 [&>*:nth-child(2n+3)]:bg-[#d1447a]"
      data-testid="Details-1"
    >
      <img
        class="h-20 w-max ml-auto mr-auto"
      />
      <div
        class="flex justify-between px-4 py-4"
      >
        <span>
          Name :
        </span>
        <span>
          <i
            class="fa-solid fa-circle-arrow-right pl-2"
          />
        </span>
      </div>
      <div
        class="flex justify-between px-4 py-4"
      >
        <span>
           Rank :
        </span>
        <span>
          <i
            class="fa-solid fa-circle-arrow-right pl-2"
          />
        </span>
      </div>
      <div
        class="flex justify-between px-4 py-4"
      >
        <span>
          Total Supply :
        </span>
        <span>
          <i
            class="fa-solid fa-circle-arrow-right pl-2"
          />
        </span>
      </div>
      <div
        class="flex justify-between px-4 py-4"
      >
        <span>
          Available Supply :
        </span>
        <span>
          <i
            class="fa-solid fa-circle-arrow-right pl-2"
          />
        </span>
      </div>
      <div
        class="flex justify-between px-4 py-4"
      >
        <span>
          Price :
        </span>
        <span>
          <i
            class="fa-solid fa-circle-arrow-right pl-2"
          />
        </span>
      </div>
      <div
        class="flex justify-between px-4 py-4"
      >
        <span>
           Price Change Weekly :
        </span>
        <span>
          <span
            class="text-green-400"
          />
          <i
            class="fa-solid fa-circle-arrow-right pl-2"
          />
        </span>
      </div>
      <div
        class="flex justify-between px-4 py-4"
      >
        <span>
          Price Change Daily :
        </span>
        <span>
          <span
            class="text-green-400"
          />
          <i
            class="fa-solid fa-circle-arrow-right pl-2"
          />
        </span>
      </div>
      <div
        class="flex justify-between px-4 py-4"
      >
        <span>
          Price Change Hourly :
        </span>
        <span>
          <span
            class="text-green-400"
          />
          <i
            class="fa-solid fa-circle-arrow-right pl-2"
          />
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
